engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oidc/OidcDiscovery.scala

```diff
@@ -0,0 +1,45 @@
+package pl.touk.nussknacker.ui.security.oidc
+
+import com.typesafe.scalalogging.LazyLogging
+import io.circe.Decoder
+import io.circe.generic.extras.{Configuration, ConfiguredJsonCodec, JsonKey}
+import net.ceedubs.ficus.readers.URIReaders
+import sttp.client.circe.asJson
+import sttp.client.{NothingT, SttpBackend, UriContext, basicRequest}
+import sttp.model.MediaType
+
+import java.net.URI
+import scala.concurrent.duration.{Duration, SECONDS}
+import scala.concurrent.{Await, ExecutionContext, Future}
+
+object OidcDiscovery extends LazyLogging {
+
+  implicit val uriDecoder: Decoder[URI] = Decoder.decodeString.map(URI.create)
+  implicit val config: Configuration = Configuration.default
+
+  def apply(issuer: URI)(implicit ec: ExecutionContext, sttpBackend: SttpBackend[Future, Nothing, NothingT]): Option[OidcDiscovery] =
+    Await.result(
+      basicRequest
+        .contentType(MediaType.ApplicationJson)
+        .get(uri"$issuer/.well-known/openid-configuration")
+        .response(asJson[OidcDiscovery]).send(),
+      Duration(30, SECONDS)
+    ).body match {
+      case Right(v) => Some(v)
+      case Left(err) =>
```

Comment:
Maybe we can handle other errors (e.g. timeout, unresolved etc.) in the same way?